app/models/subnet.rb

```diff
@@ -216,6 +220,12 @@ def unused_ip(mac = nil, excluded_ips = [])
       end
       logger.debug("Not suggesting IP Address for #{self} as no free IP found in our DB")
       return
+    elsif self.ipam == IPAM_MODES[:eui64] && self.class.supports_ipam_mode?(:eui64)
```

Comment:
Changing it in this commit for the time being would drive the design to be a bit cleaner IMO, you would also be able to get rid of `self.supports_ipam_mode?(mode)` here if `unused_ip` is implemented in the subclasses (less code to maintain). I imagine the `db` part of it will be implemented as a service if it needs to be shared between both.